addons/analytic/models/analytic_mixin.py

```diff
@@ -56,29 +62,51 @@ def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query:
 
         analytic_distribution_sql = self._field_to_sql(alias, 'analytic_distribution', query)
         value = [str(id_) for id_ in value if id_]  # list of ids -> list of string
-        if operator == 'in':  # 'in' -> ?|
+        if operator == 'in':  # keys can be comma-separated ids, we will fetch each id separately
             return SQL(
-                "%s ?| ARRAY[%s]",
-                analytic_distribution_sql,
+                "%s && %s",
+                self._query_analytic_accounts(),
                 value,
             )
         if operator == 'not in':
             return SQL(
-                "(NOT %s ?| ARRAY[%s] OR %s IS NULL)",
-                analytic_distribution_sql,
+                "(NOT %s && %s OR %s IS NULL)",
+                self._query_analytic_accounts(),
                 value,
                 analytic_distribution_sql,
             )
         raise UserError(_('Operation not supported'))
 
     def _read_group_groupby(self, groupby_spec: str, query: Query) -> SQL:
+        """To group by `analytic_distribution`, we first need to separate the analytic_ids and associate them with the ids to be counted
+        Do note that only '__count' can be passed in the `aggregates`"""
```

Comment:
Would a ValueError be more relevant? I guess it would be an issue in the code rather than the user's fault.

Reply:
Sure, it can. (I just followed the logic we had with UserError if you're doing a search with operators not supported in _condition_to_sql, but ValueError would probably be better).